src/components/archivesSort.tsx

```diff
@@ -0,0 +1,41 @@
+import { Link } from "gatsby"
+import * as React from "react"
+import ShowYear from "../components/showYear"
+
+interface ArchivesSortProps {
+  readonly index: number
+
+  readonly currentNode: {
+    frontmatter: {
+      readonly author: string[]
+      readonly title: string
+      readonly date: string
+    }
+  }
+  readonly preNode: {
+    readonly frontmatter: {
+      readonly date: string
+    }
+  }
+}
+
+export class ArchivesSort extends React.Component<ArchivesSortProps, {}> {
```

Comment:
With React components we typically match the component name with the filename 1:1 -- so `archivesSort.tsx` -> `ArchivesSort.tsx`, since components are CapsCase.

Also, rather than writing classes, we can use [function components](https://reactjs.org/docs/components-and-props.html). Classes have their use-cases, but for basic display rendering function components are preferred, since they're simpler.

Reply:
Thx for pointing out.  I read some article about "class component vs function component ".  There are a lot of benefit by using function component. I will replace all the class component to function component.